services/QuillLMS/app/services/vitally_integration/serialize_vitally_sales_user.rb

```diff
@@ -79,6 +91,18 @@ def account_data
     end
   end
 
+  private def get_from_cache(key)
+   last_school_year = Date.today.year - 1
+    cached_data = $redis.get("teacher_id:#{@user.id}_vitally_stats_for_year_#{last_school_year}")
```

Comment:
I think you can utilize your new cache object: `cached_data = CacheVitallyTeacherData.get(@teacher.id, last_school_year)`